src/components/article/CreateArticle.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import NavBar from '../presentation/NavBar';
+import Footer from '../presentation/Footer';
+import ArticleForm from './ArticleForm';
+import DisplayMessage from '../presentation/DisplayMessage';
+import { createNewArticle } from '../../actions/articles';
+
+class CreateArticle extends React.Component {
+  state = {
+    title: '',
+    body: '',
+    description: '',
+    tags: [],
+    imgUrl: '',
+    message: false,
+  }
+
+  componentDidMount() {
```

Comment:
I think componentWillMount() would be more elegant here as the user login status is readily available in props. What do you think?

Reply:
Both of them will actually work, but I understand your concern. componentWillMount makes sure it runs before the component mounts and that actually makes more sense for the task. I'll implement that and get back to you. thanks 👍

Reply:
@marcdomain @okoroemeka is right actually, see more details here -> https://reactjs.org/docs/react-component.html#unsafe_componentwillmount

Reply:
Thanks @oluwajuwon and @okoroemeka this was enlightening.